File: backend/app/schemas/report.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from datetime import datetime
from typing import Optional

class ReportCreate(BaseModel):
    inspection_id: UUID

class ReportOut(BaseModel):
    id: UUID
    inspection_id: UUID
    generated_by: UUID
    generator_name: Optional[str] = None
    inspection_title: Optional[str] = None
    minio_key: str
    sha256: str
    download_url: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: mobile/lib/shared/models/report.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'report.freezed.dart';
part 'report.g.dart';

@freezed
abstract class Report with _$Report {
  const factory Report({
    required String id,
    @JsonKey(name: 'inspection_id') required String inspectionId,
    @JsonKey(name: 'generated_by') required String generatedBy,
    @JsonKey(name: 'generator_name') String? generatorName,
    @JsonKey(name: 'inspection_title') String? inspectionTitle,
    @JsonKey(name: 'minio_key') required String minioKey,
    required String sha256,
    @JsonKey(name: 'signature_key') String? signatureKey,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @JsonKey(name: 'download_url') String? downloadUrl,
  }) = _Report;

  factory Report.fromJson(Map<String, dynamic> json) => _$ReportFromJson(json);
}
